test/clientconn_test.go

```diff
@@ -83,3 +87,131 @@ func (s) TestClientConnClose_WithPendingRPC(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+type testStatsHandler struct {
+	nameResolutionDelayed bool
+}
+
+// TagRPC is called when an RPC is initiated and allows adding metadata to the
+// context. It checks if the RPC experienced a name resolution delay and
+// updates the handler's state.
+func (h *testStatsHandler) TagRPC(ctx context.Context, rpcInfo *stats.RPCTagInfo) context.Context {
+	h.nameResolutionDelayed = rpcInfo.NameResolutionDelay
+	return ctx
+}
+
+// This method is required to satisfy the stats.Handler interface.
+func (h *testStatsHandler) HandleRPC(_ context.Context, _ stats.RPCStats) {}
+
+// TagConn exists to satisfy stats.Handler.
+func (h *testStatsHandler) TagConn(ctx context.Context, _ *stats.ConnTagInfo) context.Context {
+	return ctx
+}
+
+// HandleConn exists to satisfy stats.Handler.
+func (h *testStatsHandler) HandleConn(_ context.Context, _ stats.ConnStats) {}
+
+// TestClientConnRPC_WithoutNameResolutionDelay verify that if the resolution
+// has already happened once before at the time of making RPC, the name
+// resolution flag is not set indicating there was no delay in name resolution.
+func (s) TestClientConnRPC_WithoutNameResolutionDelay(t *testing.T) {
+	stub := &stubserver.StubServer{
+		EmptyCallF: func(context.Context, *testpb.Empty) (*testpb.Empty, error) {
+			return &testpb.Empty{}, nil
+		},
+	}
+	if err := stub.Start(nil); err != nil {
+		t.Fatalf("Failed to start StubServer: %v", err)
+	}
+	defer stub.Stop()
+
+	statsHandler := &testStatsHandler{}
+	rb := manual.NewBuilderWithScheme("instant")
+	rb.InitialState(resolver.State{Addresses: []resolver.Address{{Addr: stub.Address}}})
+	cc, err := grpc.NewClient(rb.Scheme()+":///test.server",
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithResolvers(rb),
+		grpc.WithStatsHandler(statsHandler),
+	)
+	if err != nil {
+		t.Fatalf("grpc.NewClient() failed: %v", err)
+	}
+	defer cc.Close()
+
+	cc.Connect()
+	if err := waitForReady(cc); err != nil {
+		t.Fatalf("Connection not ready: %v", err)
```

Comment:
```suggestion
		t.Fatalf("Error waiting for client to become READY: %v", err")
```

(Most important part being including "waiting for" since that's what we were doing.)

Reply:
Done